From 924d61f0752fd36fa0cccb124fb42dbb5717e1d2 Mon Sep 17 00:00:00 2001
From: midas <dev@null>
Date: Tue, 3 Jun 2025 14:46:52 +0200
Subject: [PATCH 2/3] add apiget groups command

---
 simpx/client.py  | 7 +++++++
 simpx/command.py | 5 +++++
 2 files changed, 12 insertions(+)

diff --git a/simpx/client.py b/simpx/client.py
index 5f4c94b..e3e8c06 100644
--- a/simpx/client.py
+++ b/simpx/client.py
@@ -213,6 +213,13 @@ class ChatClient:
         r = await self.send_chat_command({"type": "addressAutoAccept"})
         if r["type"] != "userContactLinkUpdated":
             raise ChatCommandError("Error changing user contact address mode", r)
+  
+    async def api_get_groups(self) -> None:
+        """get groups joined by the user"""
+        r = await self.send_chat_command({"type": "apiGetGroups"})
+        if r["type"] != "groupsList":
+            raise ChatCommandError("Error getting groups list",r)
+        return r
     
     async def api_get_chats(self, user_id: int) -> List[Chat]:
         """Get chats for a user."""
diff --git a/simpx/command.py b/simpx/command.py
index 7858219..896fea0 100644
--- a/simpx/command.py
+++ b/simpx/command.py
@@ -161,6 +161,9 @@ class SetIncognito(IChatCommand):
     type: Literal["setIncognito"]
     incognito: bool
 
+class APIGetGroups(IChatCommand):
+    type: Literal["apiGetGroups"]
+
 class APIGetChats(IChatCommand, total=False):
     type: Literal["apiGetChats"]
     userId: int
@@ -308,6 +311,7 @@ ChatCommand = Union[
     StartChat,
     APIStopChat,
     SetIncognito,
+    APIGetGroups,
     APIGetChats,
     APIGetChat,
     APISendMessage,
@@ -400,6 +404,7 @@ def cmd_string(cmd: ChatCommand) -> str:
         "startChat": lambda c: f"/_start subscribe={on_off(c.get('subscribeConnections'), False)} expire={on_off(c.get('enableExpireChatItems'), False)}",
         "apiStopChat": lambda _: "/_stop",
         "setIncognito": lambda c: f"/incognito {on_off(c['incognito'])}",
+        "apiGetGroups": lambda _: "/groups",
         "apiGetChats": lambda c: f"/_get chats pcc={on_off(c.get('pendingConnections'), False)}",
         "apiGetChat": lambda c: f"/_get chat {c['chatType']}{c['chatId']}{pagination_str(c['pagination'])}" + (f" {c['search']}" if c.get('search') else ""),
 
-- 
2.49.0

